pkg/projectfile/projectfile.go

```diff
@@ -1307,7 +1307,7 @@ func storeProjectMapping(cfg ConfigGetter, namespace, projectPath string) {
 
 // CleanProjectMapping removes projects that no longer exist
 // on a user's filesystem from the projects config entry
-func CleanProjectMapping(cfg ConfigGetter) {
+func CleanProjectMapping(cfg ConfigGetter, projectPaths ...string) {
```

Comment:
This shouldn't need projectPaths as input. We're removing duplicate entries that already exist. So just iterate through the list and when encountering a duplicate entry only keep the latest one.

Reply:
There are no duplicate entries at this point so I'm passing the path to the headless project so we can remove it. The next line adds a project entry for the newly created project.

Reply:
The duplicate entries won't be in the same list, one is in the entry for the headless project and one is in the entry for the newly created project on `push`. That said there is a bug with this approach. I think we need a clean config function that takes a `namespace` and `path` argument so we can remove the path entry from the headless project specifically.